src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -366,15 +371,34 @@ export const useBalanceChartData: UseBalanceChartData = args => {
     }
 
     // create empty buckets based on the assets, current balances, and timeframe
-    const emptyBuckets = makeBuckets({ assetIds, balances, timeframe })
+    const emptyBuckets = makeBuckets({
+      assetIds,
+      // TODO: this should be removed when defi opportunity abstractions were completed.
+      // fox farming balances are not in the Portfolio by default
+      // this hack will add the fox farming balances to the LP token balance
+      balances: {
+        ...balances,
+        [foxEthLpAssetId]: featureFlags.FoxFarming
+          ? bnOrZero(balances[foxEthLpAssetId])
+              .plus(bnOrZero(foxFarmingTotalBalanceInBaseUnit))
+              .toString()
+          : '0',
+      },
+      timeframe,
+    })
     // put each tx into a bucket for the chart
     const buckets = bucketEvents(txs, rebases, emptyBuckets)
 
     // iterate each bucket, updating crypto balances and fiat prices per bucket
     const calculatedBuckets = calculateBucketPrices({
       assetIds,
       buckets,
-      cryptoPriceHistoryData,
+      cryptoPriceHistoryData: {
+        ...cryptoPriceHistoryData,
+        // TODO: this should be removed when defi opportunity abstractions were completed.
+        // this is an ugly hack to overcome missing lp token price for charts
+        [foxEthLpAssetId]: [{ price: lpTokenPrice, date: 0 }],
```

Comment:
Not tested at runtime yet but we should be extremely cautious about `useBalanceChartsData` diffs, ideally fixing charts would not mean touching this file

Reply:
Functionally looks good both with the flag on and off but I still have concerns about this living here. Ideally we shouldn't need to change anything here, but given this doesn't live in state, the other solutions I can think of are just as hacky.

Given the fact that this is *actually* temporary and we plan to refactor this whole domain, this might be a hack we can live with for now.
@0xdef1cafe FYI ^